src/node.cc

```diff
@@ -836,6 +838,18 @@ void SetupNextTick(const FunctionCallbackInfo<Value>& args) {
   args.GetReturnValue().Set(ret);
 }
 
+void WeakCallback(
+    const v8::WeakCallbackInfo<v8::Persistent<v8::Promise,
+      node::ResetInDestructorPersistentTraits<v8::Promise> > >& data) {
+  Environment* env = Environment::GetCurrent(data.GetIsolate());
+  Local<Promise> promise = data.GetParameter()->Get(env->isolate());
+
+  Local<Value> err = promise->Result();
+  Local<Message> message = Exception::CreateMessage(env->isolate(), err);
+
+  node::InternalFatalException(env->isolate(), err, message, true);
```

Comment:
We can’t do this inside the `WeakCallback` – running JS during GC is generally a no-go

Reply:
Hm, ok. I guess I could schedule the fatal error with setImmediate in that case. Would that work?